internal/integration/rules.go

```diff
@@ -209,38 +209,47 @@ func addAttributes(targetMetrics *TargetMetrics, rules []AddAttributesRule) {
 type ignoreRules []IgnoreRule
 
 func (rules ignoreRules) shouldIgnore(name string, metricType metricType) bool {
-	var prefixesLen, metricTypesLen, exceptRulesLen int
+	if rules.isMetricExcludedFromFilters(name) {
+		return false
+	}
+
 	for _, rule := range rules {
-		exceptRulesLen += len(rule.Except)
-		for _, prefix := range rule.Except {
-			if strings.HasPrefix(name, prefix) {
-				return false
-			}
+		// if metricTypesLen or prefixesLen are not defined and exceptRulesLen
+		// is not empty then all not excepted metric should be dropped
+		totalDroppingRules := len(rule.MetricTypes) + len(rule.Prefixes)
+		if totalDroppingRules == 0 && len(rule.Except) != 0 {
+			return true
```

Comment:
is there any except rule it will ignore the metric without checking the prefix? is this ok?

Reply:
oh i get this now, at this point all `except` rules has executed, so if the metric is not excepted and there are not other rules it should be ignored.